node/core/approval-voting/src/lib.rs

```diff
@@ -779,16 +763,15 @@ where
 	}
 
 	let mut state = State {
-		session_window: None,
 		keystore: subsystem.keystore,
 		slot_duration_millis: subsystem.slot_duration_millis,
 		clock,
 		assignment_criteria,
-		db_config: subsystem.db_config,
-		db: subsystem.db,
 		spans: HashMap::new(),
 	};
 
+	// `None` on start-up. Gets initialized/updated on leaf update
+	let mut session_info_provider: Option<SessionInfoProvider> = None;
```

Comment:
We can drop the `Option` if it's safe to block the subsystem until we get the first `ActiveLeavesUpdate`.

Reply:
No need to block. `RuntimeInfo` does not need an active leaf for [creation](https://github.com/paritytech/polkadot/blob/a7b74ccbfd520146765e20cca9641f7e4ae8e213/node/subsystem-util/src/runtime/mod.rs#L105).